src/Qoollo.PerformanceCounters/InternalCounters/Counters/InternalDeltaCounter.cs

```diff
@@ -0,0 +1,178 @@
+            }
+        }
+
+        /// <summary>
+        /// Метод создания дескриптора для InternalNumberOfItemsCounter
+        /// </summary>
+        /// <param name="name">Имя счётчика</param>
+        /// <param name="description">Описание счётчика</param>
+        /// <returns>Дескриптор</returns>
+        internal static InternalCounterDescriptor CreateDescriptor(string name, string description)
+        {
+            return new Descriptor(name, description);
+        }
+
+        // ===================
+
+        /// <summary>
+        /// Конструктор InternalNumberOfItemsCounter
+        /// </summary>
+        /// <param name="name">Имя счётчика</param>
+        /// <param name="description">Описание счётчика</param>
+        public InternalDeltaCounter(string name, string description)
+            : base(name, description)
+        {
+            _currentValue = 0;
+        }
+
+        /// <summary>
+        /// Уменьшить значение на 1
+        /// </summary>
+        /// <returns>Текущее значение или Counter.FailureNum</returns>
+        public override long Decrement()
+        {
+            return Interlocked.Decrement(ref _currentValue);
+        }
+
+        /// <summary>
+        /// Уменьшить значение на "value"
+        /// </summary>
+        /// <param name="value">Значение, на которое уменьшаем</param>
+        /// <returns>Текущее значение или Counter.FailureNum</returns>
+        public override long DecrementBy(long value)
+        {
+            return Interlocked.Add(ref _currentValue, -value);
+        }
+
+        /// <summary>
+        /// Увеличить значение на 1
+        /// </summary>
+        /// <returns>Текущее значение или Counter.FailureNum</returns>
+        public override long Increment()
+        {
+            return Interlocked.Increment(ref _currentValue);
+        }
+
+        /// <summary>
+        /// Увеличить значение на "value"
+        /// </summary>
+        /// <param name="value">Значение, на которое увеличиваем</param>
+        /// <returns>Текущее значение или Counter.FailureNum</returns>
+        public override long IncrementBy(long value)
+        {
+            return Interlocked.Add(ref _currentValue, value);
+        }
+
+        /// <summary>
+        /// Задать значение
+        /// </summary>
+        /// <param name="value">Новое значение</param>
+        public override void SetValue(long value)
+        {
+            Interlocked.Exchange(ref _currentValue, value);
+        }
+
+        /// <summary>
+        /// Получает текущее значение счетчика производительности
+        /// </summary>
+        public override long CurrentValue => Interlocked.Read(ref _currentValue);
+
+        /// <summary>
+        /// Take next sample interval and return its value
+        /// </summary>
+        /// <returns>Difference between counter raw value during sample interval</returns>
+        public override long Measure()
+        {

```

Comment:
I checked my solution one more time and found that it has one small problem. 
Suppose there is only one thread that updates counter in a cycle by calling ```Increment()```. In this situation any call to ```Measure()``` should never return negative value, but in my implementation it can.
Scenario:
1. Thread 1 enters ```Measure()``` and read ```_currentValue``` == __X__
2. Updating thread calls ```Increment()```
3. Thread 2 enters ```Measure()``` and read ```_currentValue``` == __X + 1__
4. Thread 2 reads and updates atomially ```_lastMeasuredValue``` so it becomes __X + 1__ and returns the correct value
5. Thread 1 reads and updates atomially ```_lastMeasuredValue``` and observe that it is equal to __X + 1__ whereas ```_currentValue``` was read equal to __X__. As a result the thread will return __-1__

To prevent such scenario all threads that call ```Measure()``` should be synchronized with lightweight lock that has already been implemented by you.

So the correct implementation would be:
```C#
public override long Measure()
{
    long result = 0;
    try { }
    finally 
    {
        var sw = new SpinWait();
        while (!TryTakeLock())
            sw.SpinOnce();

        long currentValue = Interlocked.Read(ref _currentValue);
        long lastMeasuredValue = Interlocked.Exchange(ref _lastMeasuredValue, currentValue);
        result = currentValue - lastMeasuredValue;

        ReleaseLock();
    }
    return result;
}
```

Reply:
Sure, it was my thinking also, so I just wanted to lock those statements to critical section. Using your code.